core/kazoo_documents/src/kzd_cluster.erl

```diff
@@ -0,0 +1,23 @@
+%%%-----------------------------------------------------------------------------
+%%% @copyright (C) 2019, 2600Hz
+%%% @doc Accessors for cluster document
+%%% @author Kalyan Krause
+%%% @end
+%%%-----------------------------------------------------------------------------
+-module(kzd_cluster).
+
+-define(MOD_CONFIG_CLUSTER, <<"cluster">>).
+
+-export([id/0]).
+
+-include("kz_documents.hrl").
+
+-spec id() -> kz_term:ne_binary().
+id() ->
+    case kapps_config:get_ne_binary_value(?MOD_CONFIG_CLUSTER, <<"cluster_id">>) of
+        'undefined' ->
+            ClusterId = kz_binary:rand_hex(16),
+            {'ok', _JObj} = kapps_config:set_default(?MOD_CONFIG_CLUSTER, <<"cluster_id">>, ClusterId),
+            ClusterId;
+        ClusterId -> ClusterId
+    end.
```

Comment:
does `kapps_config:get_ne_binary_value(?MOD_CONFIG_CLUSTER, <<"cluster_id">>, kz_binary:rand_hex(16), <<"default">>)` not work ? i think it will update and return it if its not created.

Reply:
Possibly, I just modified already existing code.
It also looks like it's `kapps_config:get_ne_binary`.

Reply:
you're correct, that's a  `kz_json` thing

Reply:
What is last parameter (`<<"default">>`) in that call for?

Reply:
it's the node, otherwise you'll get/set from node()